src/EventStore.Core/TransactionLog/Chunks/TFChunkWriter.cs

```diff
@@ -41,43 +43,80 @@ public class TFChunkWriter : ITransactionFileWriter {
 		}
 
 		public bool Write(ILogRecord record, out long newPos) {
+			OpenTransaction();
+			bool wasWritten = WriteToTransaction(record, out newPos);
+			CommitTransaction(flush: !wasWritten);
```

Comment:
i dont think we need to flush here. this is for the case where we try to write but the chunk is full. before this PR that would cause a new chunk to be written and the writer checkpoint to be flushed at the start of the new chunk. now though we can leave it unflushed because we'll land in the same case a partially completed transaction: with the writer checkpoint in a previous chunk and the startup routine detecting and removing the new chunk(i write all this to explain it to future tim rather than to explain it to you)

then we can take the Flush call out of Write and just let the two places that want to flush call flush themselves and keep them separate

Reply:
fixed :+1: i've also adjusted the commit message to describe the change in behaviour (although I've not added the details)